[WIP] Begin homebrew version experiment

ChrisChinchilla opened the pull request:
@erak and @chriseth I made a little experiment at supporting versions of formulae, so this works by installing (in this example):

```
brew install solidity
```

for latest version, and:

```
brew install solidity@5.2 # add .rb to test locally before it's published
```

for 5.2 etc. I got this working, but need to figure out how to alias `solc` etc to something like `solc52`, so you can specify the version, this is how Python versions does it with `pip` for example.

I can keep plugging at it, or we can link into the Github issues where people have asked for this feature and they can also help out :)

chriseth commented:
Actually why don't we have CircleCI tests for this repository? It could install all packages defined here one after the other and check that the binaries satisfy some basic tests.

ChrisChinchilla commented:
@chriseth I am happy to try that, but will need the right access :)

chriseth commented:
Ok: https://github.com/ethereum/homebrew-ethereum/pull/214

chriseth commented:
https://github.com/ethereum/homebrew-ethereum/pull/214 is merged, so if you rebase this one, it will automatically test the formula.

axic commented:
#214 actually won't test the formula because you need to refer to it with `solidity@5.2.rb`

chriseth commented:
Ah and it is actually a different file. Can we put all formulas in the same file?

axic commented:
> Can we put all formulas in the same file?

I think homebrew doesn't support that :(

ChrisChinchilla commented:
Yeah @chriseth and @axic I also find the requirement for different versions in different files messy, but it's the way it is. I'll get back to this later in the week for some more testing and polishing.

ChrisChinchilla commented:
@chriseth and @axic I changed this slightly to allow people to install 0.5.x from default homebrew formula, and then this formula to support the last version of 0.4.x. As part of this I added an alternative boost version that fixes builds for this version, and allows users to have different versions of boost installed too.

I feel like these two versions are enough, but we can add specific ones if people ask now we know how to do it.

ChrisChinchilla commented:
The next thing I want to figure out is how to enable people to have multiple versions of Solidity installed, and they can then use `solc` and `solc4` for example, which is how Python does it with pip. But I think we can add that later.

zulhfreelancer commented:
It's time to have Solidity Version Manager (SVM) I guess.

ChrisChinchilla commented:
Closing in favour of https://github.com/ethereum/homebrew-ethereum/pull/221

dguido commented:
@zulhfreelancer Here you go: https://github.com/crytic/solc-select
